### DISCUSSION TITLE
Maintain math mode markup in title when importing from clipboard

### DISCUSSION THREAD

**Discussion by raymondEhlers** [retorquere/zotero-better-bibtex#2724] [Q&A] [answered]
Hi,

Thanks for this nice plugin! I'm having some difficulties with maintaining the latex math markup on import. For my example, I'm trying to import:

```
@article{ALICE-PUBLIC-2018-014,
      collaboration = "ALICE",
      title         = "{ALICE 2017 luminosity determination for pp collisions at
                       $\sqrt{s}$ = 5 TeV}",
      year          = "2018",
      url           = "https://cds.cern.ch/record/2648933",
}
```

My goal is just to pass through the `$\sqrt{s}$` so that it will be rendered properly when used for citations - I don't mind how it shows up in Zotero. (I recognize this is limited in terms of fields, but it's a simple way to reproduce something we frequently deal with). I've tried all of the options I can find, including every permutation of modifying the title I can find (eg. case-protection, sentence case, export as unicode or not), as well as varying the `importUnknownTexCommand` setting to either `tex` or `text`, but I can't find the right combination to pass this through. Depending on how I configure the options, I can get outputs like:
![Screenshot 2023-12-11 at 3 25 45 PM](https://github.com/retorquere/zotero-better-bibtex/assets/1571927/8bfb1acf-ab7d-4235-8aaa-d60d8b256a23)

Unfortunately, none of those are correct. I just want it to be stored as `ALICE 2017 luminosity determination for pp collisions at $\sqrt{s}$ = 5 TeV`. I've used the postscript for modifying exported values, but it seems as if the import parser is a bit too aggressive and stops before reaching the next `$`. Is there any way to tell the importer to leave everything exactly as is? Thanks!

**Comment by retorquere** [comment]
You can disable BBT import in the preferences, it will fall back to the standard Zotero BibTeX import, which may leave more of this in place.

**Reply by raymondEhlers** [reply]
Thanks for the suggestion! Unfortunately, the standard Zotero BiBTeX also doesn't handle this properly - it returns `ALICE 2017 luminosity determination for pp collisions at \sqrts = 5 TeV`, so it removes both the brackets and the `$`. This sort of standard Zotero behavior in the export was why I started using BBT, which was worked nicely to resolve issues in the export. I also note that if I import from eg. arXiv, everything is carried through correctly, so somehow it's about how the latex is interpreted on import.

I think I forgot a case above (or rather, should not have taken a screenshot) - if I turn all of all options to modify the title (don't encode was unicode, don't apply title casing, etc), I get `ALICE 2017 luminosity determination for pp collisions at <span class="nocase">s</span> = 5 TeV`, where it drops the `\sqrt` entirely, which seems like a potential bug? Is it somehow possible to customize the import similar to the export? Perhaps it's not possible, but my naive expectation from reading documentation was that it should be possible to pass the title through entirely as is.

If I set `importUnknownTexCommand`, I can get `ALICE 2017 luminosity determination for pp collisions at <span class="nocase"><script>\sqrt</script>s</span> = 5 TeV`, which is the closest that I've gotten, but unfortunately that still isn't workable

**Reply by retorquere** [reply]
> where it drops the \sqrt entirely, which seems like a potential bug? 

For bug diagnosis you will have to open an issue.

> it should be possible to pass the title through entirely as is.

it isn't, currently.

**Reply by raymondEhlers** [reply]
Okay, thanks! I wanted to confirm that I wasn't overlooking an option before opening an issue. Since it sounds like I'm not, I'll document things a bit more carefully and open issue.

Thanks again for your quick responses!

**Comment by Claudio-Agostini** [comment, marked as answer]
Please correct me if I'm wrong, but now the option has been added: going to "Zotero preferences -> Advanced -> Config Editor" there is the option "extensions.zotero.translators.better-bibtex.rawImports" that if turned on allows importing titles from Bibtex without changing anything.
(It also marks the item automatically with "#LaTeX" which allows to export it again in the correct way).

**Reply by retorquere** [reply]
> My goal is just to pass through the $\sqrt{s}$ so that it will be rendered properly when used for citations

I had interpreted this as having it render usefully in Word. If that's not needed, raw entries could work.

**Reply by raymondEhlers** [reply]
Yes, indeed - by "rendered properly", I meant passed through unmodified so that bibtex+latex could do the rendering itself. Apologies for my lack of clarity!

I just ran some tests, and it appears that `rawImports` addresses this issue. I seem to recall trying that option at the time and running into an issue with the output - it must have been some other settings (maybe on export? My notes from the time are unclear on this particular point), as it seems to be working end to end now. Thanks!
